### ISSUE TITLE
Add an exact match rule to the template loader

### ISSUE THREAD

**Issue by alvarofranz** [winnipress/winnipress#3]
In many cases I had to create a new blank page in the WP Admin just to be able to add a `page-whatever.php` in my theme folder.

Also, creating a template for a subdirectory like `example.com/whatever/something` is not a trivial task in WP.

The idea is to check before loading any template if there is a PHP file in the theme folder that matches the requested path.

This way, if we want to do some custom stuff in our `example.com/whatever` page, we don't need to create a blank page. We just throw in a `whatever.php` file into the theme folder.

## Prefixes to avoid confusion

The matching file should be prefixed to recognize it just like currently we do `single-whatever.php`, `page-whatever.php`, ...

The prefix could be `path-whatever.php`

## What happens with subfolders

Yet to think of a good idea.

`path-whatever--something.php` sounds not bad to me.

Also `path-whatever/something.php` seems a nice approach.

**Comment by joyously** [comment]
Can I just say `_doing_it_wrong()` ?
Theme templates handle the standard queries. For special stuff, you should have a plugin, and that plugin could add a rewrite rule for the special page, supply the special handling, and filter the `template_include` or use `template_redirect` to get to the special handling of an entire page. 
To keep the special page looking like the rest of the pages, the plugin would supply it via a shortcode, which would mean there is an actual page created (by user or plugin) that can be edited.

**Comment by alvarofranz** [comment]
@joyously I really thank you for your feedback. But the idea of this fork is to actually get aligned with the things that I miss from WordPress or ClassicPress.

I almost never use pages at all, I just create them in the theme, add some custom content and logic and there it goes.

So this idea is a step forwards into what I need. I don't want to create a page in de admin just to give it a title and insert some shortcodes. This will also require requesting the database just to get the page title and the shortcodes that then will also have to be parsed... that's insane.

I want to be able to create the page and add WP functionality into it together with some Vue or React components and expect it to work without any extra fuss.

The idea of WinniPress is to get closer to a development framework that makes development easier from a coding point of view and run away from buttons and page builders.

**Comment by joyously** [comment]
Yes, I know the purpose. But it can be a big hassle to keep the pages consistent if the theme is not involved.
It seems like a step backwards, into how you would do a site without a CMS.

**Comment by alvarofranz** [comment]
@joyously You are right. It will be a problem if the theme contains a "Contact" page and I also create a "Contact" page in the backend.

Also, it's somehow taking a step backwards in that sense. But as I see it the big potential of the CMS tools is not in pages but in posts (whatever kind of custom post type). Since that is really dynamic content that will be created/deleted/modified...

But pages are a static thing. If I need a Contact page I don't want it to be dynamic, I want to develop it once, throw it into the theme and expect it to work when visited. But without the framework having to fetch the database to make sure that page exists or get it's title.

But yes you provided a strong point, and I will think about alternatives and if the road I am going makes sense at all. In the end this is about trying to create something for fun and also that will work and make me happy when developing projects.
